packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol

```diff
@@ -125,12 +125,14 @@ contract OptimisticAssertor is Lockable, OptimisticAssertorInterface, Ownable {
 
     function getAssertion(bytes32 assertionId) public view returns (bool) {
         Assertion memory assertion = assertions[assertionId];
+        // Return early if not using answer from resolved dispute.
+        if (assertion.disputer != address(0) && !assertion.useDisputeResolution) return false;
         require(assertion.settled, "Assertion not settled"); // Revert if assertion not settled.
         return assertion.settlementResolution;
     }
 
     function settleAndGetAssertion(bytes32 assertionId) public returns (bool) {
-        settleAssertion(assertionId);
+        if (!assertions[assertionId].settled) settleAssertion(assertionId);
```

Comment:
This is to make it consistent behavior of Optimistic Oracle `settleAndGetPrice` that can be called repeatedly and skipping settlement if already done.